src/Resources/public/reference-entity/record/table/value/value-updater.ts

```diff
@@ -34,6 +37,32 @@ export default class ValueUpdater {
         }
     }
 
+    public createDragOver(): (event: React.DragEvent<HTMLTableRowElement>) => void {
+        return (event: React.DragEvent<HTMLTableRowElement>) => {
+            event.preventDefault();
+        };
+    }
+
+    public createDragStart(index: number): (event: React.DragEvent<HTMLTableRowElement>) => void {
+        return () => {
+            this.dragPosition = index;
+        };
+    }
+
+    public createDrop(index: number): (event: React.DragEvent<HTMLTableRowElement>) => void {
+        return (event: React.DragEvent<HTMLTableRowElement>) => {
+            event.preventDefault();
+
+            const row: TableDataRow | undefined = this.internalDataRows.splice(this.dragPosition, 1).shift();
+            if (row === undefined) {
+                return;
+            }
+
+            this.internalDataRows.splice(index, 0, row);
+            this.updateDataRows(this.internalDataRows);
```

Comment:
`this.internalDataRows` was mutated before on line 56. so here its behaviour could be unexpected

Reply:
Better I create a copy of the array on 56, but I still would like to have a test to prevent a regression. So if you can provide one that would be helpful.

Reply:
sorry, i must know the data flow within the code to help you. all i see is just raw code and potential bugs in it. 
maybe this mutation is intended to be here, i just don't know, still, it is better to omit mutations in my opinion. 
try to follow all cases where you use/change `this.internalDataRows`

Reply:
`this.internalDataRows` contains the state of the table rows before the react state gets updated. The method `ValueUpdater::updateDataRows()` contains the call on `this.onChange(newValue);` which updates the react state of the component.